tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,169 @@
+
+    /* 8. Add the assertion of the in the input field is in the format "2014/05/02" */
+    await expect(page.locator('.mat-datepicker-input')).toHaveValue(`${expectedYear}/${expectedMonth}/${expectedDayLong}`)
+
+    /* 9. Select the type of pet "dog" and click "Save Pet" button */
+    await page.selectOption('select', 'dog')
+    await page.getByText('Save Pet').click()
+
+    /* 10. On the Owner Information page, add assertions for the newly created pet. Name is Tom, Birth Date is in the format "2014-05-02", Type is dog */
+
+    // Select the most recent pet record with pet information only
+    const newPetTable = page.locator('app-pet-list').last().locator('dl').first()
+
+    // Assert that the new pet's name, birthdate, and type are correct
+    await expect(newPetTable.locator('dd').first()).toHaveText('Tom')
+    await expect(newPetTable.locator('dd').nth(1)).toHaveText(`${expectedYear}-${expectedMonth}-${expectedDayLong}`)
+    await expect(newPetTable.locator('dd').nth(2)).toHaveText('dog')
+
+    /* 11. Click "Delete Pet" button the for the new pet "Tom" */
+    // Clean up from test
+    await newPetTable.getByText('Delete').click()
+
+    /* 12. Add assertion that Tom does not exist in the list of pets anymore */
+    await expect(newPetTable.locator('dd').first()).not.toHaveText('Tom')
+
+})
+
+test('Select the desired date in the calendar - 2', async ({page}) => {
+
+    /* 2. In the list of the Owners, locate the owner by the name "Jean Coleman" and select this owner */
+    await page.getByText('Jean Coleman').click()
+
+    /* 3. In the list of pets, locate the pet with a name "Samantha" and click "Add Visit" button */
+    // Create two new vet visits for the pet Samantha
+    const petRecord = page.locator('app-pet-list', {hasText: 'Samantha'})
+    await petRecord.getByText('Add Visit').click()
+
+    /* 4. Add the assertion that "New Visit" is displayed as header of the page */
+    await expect(page.locator('h2')).toHaveText('New Visit')
+
+    /* 5. Add the assertion that pet name is "Samantha" and owner name is "Jean Coleman" */
+    // Assert that the pet name and owner are correct
+    await expect(page.locator('table').first().locator('td').first()).toHaveText('Samantha')
+    await expect(page.locator('table').first().locator('td').last()).toHaveText('Jean Coleman')
+
+    /* 6. Click on the calendar icon and select the current date in date picker */
+    // Add an appointment for today's date for 'dermatology'
+    await page.locator('.mdc-icon-button').click()
+    await page.locator('.mat-calendar-body-today').click()
+
+    let date = new Date()
+    const appointmentYear1 = date.getFullYear()
+    const appointmentMonth1 = date.toLocaleString('En-US', {month: '2-digit'})
+    const appointmentDay1 = date.toLocaleString('En-US', {day: '2-digit'})
+    const expectedAppointmentDate1 = `${appointmentYear1}-${appointmentMonth1}-${appointmentDay1}`
```

Comment:
Use a better naming. Adding an indexes to a constant names is not a good practice.